src/snmalloc/global/memcpy.h

```diff
@@ -179,6 +178,142 @@ namespace snmalloc
     }
   };
 
+  /**
+   * StrictProvenance architectures are prickly about their pointers.  In
+   * particular, they may not permit misaligned loads and stores of
+   * pointer-sized data, even if they can have non-pointers in their
+   * pointer registers.  On the other hand, pointers might be hiding anywhere
+   * they are architecturally permitted!
+   */
+  struct GenericStrictProvenance
+  {
+    static_assert(bits::is_pow2(sizeof(void*)));
+    /*
+     * It's not entirely clear what we would do if this were not the case.
+     * Best not think too hard about it now.
+     */
+    static_assert(alignof(void*) == sizeof(void*));
+
+    static constexpr size_t LargestRegisterSize = 16;
+
+    static SNMALLOC_FAST_PATH_INLINE void
+    copy(void* dst, const void* src, size_t len)
+    {
+      size_t src_misalign =
+        address_misalignment<alignof(void*)>(address_cast(src));
+
+      /*
+       * As a function of misalignment relative to pointers, how big do we need
+       * to be such that the span could contain an aligned pointer?  We'd need
+       * to be big enough to contain the pointer and would need an additional
+       * however many bytes it would take to get us up to alignment.  That is,
+       * (sizeof(void*) - src_misalign) except in the case that src_misalign is
+       * 0, when the answer is 0, which we can get with some bit-twiddling.
+       */
+      size_t min_contains_aligned_ptr_size = sizeof(void*) +
+        (static_cast<size_t>(-static_cast<ptrdiff_t>(src_misalign)) &
+         (alignof(void*) - 1));
+
+      /*
+       * If this is a small size, use a jump table
+       */
+      if (len < sizeof(void*))
+      {
+        small_copies<sizeof(void*) - 1, sizeof(long)>(dst, src, len);
+      }
```

Comment:
Could you do a larger jump table here?
```suggestion
      if (len < min_contains_aligned_ptr_size)
      {
        small_copies<sizeof(void*) - 1, sizeof(long)>(dst, src, len);
      }
```